pkg/flow/events.go

```diff
@@ -1005,3 +1015,175 @@ func (events *events) listenForEvents(ctx context.Context, im *instanceMemory, c
+	if err != nil {
+		return resp, err
+	}
+
+	if jsCode, ok := cache[key]; ok {
+		script = jsCode
+
+	} else {
+		ceventfilter, err := ns.QueryCloudeventfilters().Where(enteventsfilter.NameEQ(filtername)).Only(ctx)
+		if err != nil {
+			return resp, err
+		}
+
+		script = ceventfilter.Jscode
+
+	}
+
+	var mapEvent map[string]interface{}
+	err = json.Unmarshal(cloudevent, &mapEvent)
+
+	if err != nil {
+		return resp, err
+	}
+
+	//create js runtime
+	vm := goja.New()
+
+	vm.Set("event", mapEvent)
+
+	_, err = vm.RunString(script)
+	if err != nil {
+		//panic(err)
+		return resp, err
+	}
+
+	var fn func() map[string]interface{}
+	err = vm.ExportTo(vm.Get("filter"), &fn)
+	if err != nil {
+		//panic(err)
+		return resp, err
+	}
+
+	newEventMap := fn()
+	newBytesEvent, err := json.Marshal(newEventMap)
+	if err != nil {
+		return resp, err
+	}
+
+	resp.Event = newBytesEvent
+
+	if string(resp.GetEvent()) == "null" {
+		event, _ := EventByteToCloudevent(cloudevent)
+		if err != nil {
+			return resp, err
+		}
+		flow.logToNamespace(ctx, time.Now(), ns, "Dropped Event: %s", event.ID())
+	} else {
+		event, _ := EventByteToCloudevent(newBytesEvent)
+		if err != nil {
+			return resp, err
+		}
+		flow.logToNamespace(ctx, time.Now(), ns, "Eventfilter applied, new Event : %s (%s / %s)", event.ID(), event.Type(), event.Source())
+	}
+
+	return resp, err
+}
+
+func (flow *flow) DeleteCloudEventFilter(ctx context.Context, in *grpc.DeleteCloudEventFilterRequest) (*emptypb.Empty, error) {
+
+	var resp emptypb.Empty
+
+	namespace := in.GetNamespace()
+	filtername := in.GetFilterName()
+
+	ns, err := flow.getNamespace(ctx, flow.db.Namespace, namespace)
+	if err != nil {
+		return &resp, err
+	}
+
+	_, err = ns.QueryCloudeventfilters().Where(enteventsfilter.NameEQ(filtername)).Only(ctx)
+	if err != nil {
+		err = errors.New("event filter not existing")
+		return &resp, err
+	}
+
+	_, err = flow.db.CloudEventFilters.
+		Delete().
+		Where(
+			enteventsfilter.And(
+				enteventsfilter.NameEQ(filtername),
+			)).
+		Exec(context.Background())
+
+	if err != nil {
+		return &resp, err
+	}
+
+	var key keyPair
+	key.filtername = filtername
+	key.namespace = namespace
+	delete(cache, key)
+
+	return &resp, err
+
+}
+
+func (flow *flow) CreateCloudEventFilter(ctx context.Context, in *grpc.CreateCloudEventFilterRequest) (*emptypb.Empty, error) {
+
+	var resp emptypb.Empty
+
+	namespace := in.GetNamespace()
+	filtername := in.GetFiltername()
+	script := in.GetJsCode()
+
+	script = fmt.Sprintf("function filter() {\n %s \n}", script)
+
+	//compiling js code is needed
+	_, err := goja.Compile("filter", script, false)
+	if err != nil {
+		err := errors.New("js code does not compile")
+		return &resp, err
+	}
+
+	ns, err := flow.getNamespace(ctx, flow.db.Namespace, namespace)
+	if err != nil {
+		return &resp, err
+	}
+
+	ceventfilter, _ := ns.QueryCloudeventfilters().Where(enteventsfilter.NameEQ(filtername)).Only(ctx)
```

Comment:
Don't suppress this error.

Reply:
in this case  I only want to check if an elem is in database then returning an error, so agains checking for err == nil, I thought it's better to test if an elem was found or not with if ceventfilter != nil

Reply:
The problem is that `ceventfilter` would be `nil` if any type of error occured. This includes errors you don't expect, like network connection issues. That's not a big deal because if the network is throwing errors you can't handle that and you shouldn't proceed. BUT, I really hate when error messages are misleading. It can waste a lot of time debugging the wrong problem. So if network issues happen, I want to be told about network issues, not a misleading "cloud event filter already exist". 

Instead of `.Only(ctx)`, I believe there's a `.Count(ctx)` option. Then you can check the error, and afterwards you can check that the count is zero, like you're expecting.